### PR TITLE
Raise exception class where appropriate for brevity (#2040)

### PR DESCRIPTION
**Pull request by natanlao**
To find applicable exceptions:

```console
$ ack 'raise .+\(\)'
```

To replace the most common case:

```console
$ ack -l --print0 'raise .+\(\)' | xargs -0 -n 1 sed -i '' 's/Error()/Error/g'
```

I fixed the remaining 3-4 cases manually.

---

Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `HCA` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] Added `chain` label <sub>or this PR is not the base of another PR</sub>
- [x] Made this PR a blocker of next PR in chain <sub>or this PR is not the base of another PR</sub>
- [x] Ran `make update_requirements` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or commented that sandbox will be skipped</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit

Primary reviewer (after pushing merge commit)

- [ ] Moved linked issue to Merged
- [ ] Pushed merge commit to Github
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Unassign reviewer from PR

### CODE REVIEW

**Comment by natanlao** [comment]
> Found one instance of raise that slipped through your regex

You're right that it was too strict -- I've updated the PR description and checked again.

> Also, I feel the documentation added to CONTRIBUTING.rst fits better at the end of section 1.5 Code Hygiene instead of 1.6 Type Hints. Not certain about this however so feel free to ask for a 3rd opinion if you disagree.

Oops, I definitely intended to put it under Code Hygiene. Fixed.

**Comment by hannes-ucsc** [comment]
Note that PyCharm has global search and replace with regexes:

image

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -430,6 +430,11 @@ Code Hygiene
   by it would be conflated with the one raised by d['x']. The latter is
   expected, the former usually consitutes a bug.
 
+* When raising an exception without arguments, we prefer raising the class
+  instead of constructing an instance without arguments::
```
Comment (hannes-ucsc) [inline CONTRIBUTING.rst:434]:
```suggestion
  instead of raising an instance constructed without arguments::
```

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -430,6 +430,11 @@ Code Hygiene
...
+
+    raise RuntimeError()  # verbose
```
Comment (hannes-ucsc) [inline CONTRIBUTING.rst:436]:
```suggestion
    raise RuntimeError()  # bad
```

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -430,6 +430,11 @@ Code Hygiene
...
+    raise RuntimeError    # concise
```
Comment (hannes-ucsc) [inline CONTRIBUTING.rst:437]:
```suggestion
    raise RuntimeError
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
No bump.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**